src/EventListener/TwoFactorLoginSuccessHandler.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\EventListener;
+
+use Sonata\UserBundle\GoogleAuthenticator\Helper;
+use Sonata\UserBundle\Model\User;
+use Sonata\UserBundle\Model\UserManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+
+class TwoFactorLoginSuccessHandler implements AuthenticationSuccessHandlerInterface
```

Comment:
final?

Reply:
@core23 i'm not sure if this class needs to be final

Reply:
We should make everything final by default and lower this if there are some good reasons. This allow us to make some changes without breaking BC.

Reply:
@core23 ok, done.